src/clients/SpokePoolClient.ts

```diff
@@ -354,16 +371,21 @@ export class SpokePoolClient extends BaseAbstractClient {
       return { unfilledAmount: outputAmount, fillCount: 0, invalidFills: [] };
     }
 
-    const { validFills, invalidFills } = fillsForDeposit.reduce(
-      (groupedFills: { validFills: Fill[]; invalidFills: Fill[] }, fill: Fill) => {
+    const { validFills, invalidFills, unrepayableFills } = fillsForDeposit.reduce(
+      (groupedFills: { validFills: Fill[]; invalidFills: Fill[]; unrepayableFills: Fill[] }, fill: Fill) => {
         if (validateFillForDeposit(fill, deposit).valid) {
+          if (this.isEvmRepaymentValid(fill, deposit)) {
```

Comment:
Should this be `if (!this.isEvmRepaymentValid(fill, deposit))`?

Reply:
https://github.com/across-protocol/sdk/pull/853/commits/6a7d527829360d11554faf52d5f1d1285795c84c